src/wp-includes/class-wp-query.php

```diff
@@ -2532,11 +2579,21 @@ public function get_posts() {
 					}
 
 					$orderby_array[] = $parsed;
+
+					// Check if this field needs deterministic ordering
+					if ( in_array( $orderby, $fields_requiring_deterministic_orderby, true ) ) {
+						$needs_deterministic_orderby = true;
+					} elseif ( 'ID' === $orderby ) {
+						$has_id_orderby = true;
+					}
 				}
 				$orderby = implode( ' ' . $query_vars['order'] . ', ', $orderby_array );
 
 				if ( empty( $orderby ) ) {
-					$orderby = "{$wpdb->posts}.post_date " . $query_vars['order'];
+					$orderby = "{$wpdb->posts}.post_date " . $query_vars['order'] . ', ' . "{$wpdb->posts}.ID " . $query_vars['order'];
+				} elseif ( $needs_deterministic_orderby && ! $has_id_orderby ) {
+					// Add ID as tie-breaker for deterministic ordering
+					$orderby .= ", {$wpdb->posts}.ID " . $query_vars['order'];
```

Comment:
TODO - don't forget search below. Might need it as well.


See https://github.com/WordPress/wordpress-develop/pull/1762/files#diff-1d050668621cbc3028e027d15e68438f4d879dd6e80ba98f9c9f69b9fb5469d0R2425